src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/WatchableReadStream.cs

```diff
@@ -106,6 +129,25 @@ public override int ReadByte()
             return read;
         }
 
+        public override void Close()
+        {
+            base.Close();
+            Log();
+        }
+
+        private void Log()
+        {
+            if (_logged || _logger == null || _blob == null)
+            {
+                return;
+            }
+
+            _blobAccessLog["ElapsedTimeOnRead"] = _timeRead.Elapsed;
+            _blobAccessLog["BytesRead"] = _countRead;
+            _logger.LogDebug($"ReadStream: {_blobAccessLog}");
+            _logged = true;
+        }
```

Comment:
Also, for these read/write logs, you need to add tests to the existing WatchableReadStreamTests/WatchableCloudBlobStreamTests suites, verifying the log entries